src/alm/alm_optimizer.rs

```diff
@@ -15,6 +15,26 @@ const DEFAULT_INFEAS_SUFFICIENT_DECREASE_FACTOR: f64 = 0.1;
 const DEFAULT_INITIAL_TOLERANCE: f64 = 0.1;
 const SMALL_EPSILON: f64 = std::f64::EPSILON;
 
+/// Internal/private structure used by method AlmOptimizer.step
+/// to return some minimal information about the inner problem
+struct InnerProblemStatus {
```

Comment:
We should probably add `#[derive(Debug, Copy, Clone)]` for when purple want to help contribute and start printing :)

Reply:
This structure is not public. It's only used within `AlmOptimizer`.

Reply:
What happens in you do `println!("{#?}", alm_optimizer)`?
I think you will get errors on debug not being implemented, but maybe not.

Not sure if we test this?

Reply:
`AlmOptimizer` doesn't hold instances of or references to `InnerProblemStatus` - this is just returned by method `step`. However, `AlmOptimizer` has an instance of `AlmProblem` which is in turn has fields that need to satisfy the trait constraint:

```rust
MappingAlm: Fn(&[f64], &mut [f64]) -> Result<(), SolverError>
```

therefore, as far as I know, `Debug` cannot be derived. However, if someone would like to print some meaningful information, they should print `AlmCache`, which contains all information used at every iteration of the algorithm (both for the outer and the inner loops).